lib/src/migrators/division.dart

```diff
@@ -0,0 +1,208 @@
+      withContext(true, _expectsNumericResult, () {
+        if (shouldMigrate(expression)) {
+          addPatch(patchBefore(node, "divide"));
+          patchSlashToComma(expression);
+        }
+        super.visitBinaryOperationExpression(expression);
+      });
+      return;
+    }
+    withContext(true, _expectsNumericResult,
+        () => super.visitParenthesizedExpression(node));
+  }
+
+  /// Allows division within this return rule.
+  @override
+  void visitReturnRule(ReturnRule node) {
+    withContext(true, _expectsNumericResult, () => super.visitReturnRule(node));
+  }
+
+  /// Allows division within this variable declaration.
+  @override
+  void visitVariableDeclaration(VariableDeclaration node) {
+    withContext(true, _expectsNumericResult,
+        () => super.visitVariableDeclaration(node));
+  }
+
+  /// Returns true if we assume that [operator] always returns a number.
+  ///
+  /// This is always true for * and %, and it's true for + and - as long as the
+  /// aggressive option is enabled.
+  bool returnsNumbers(BinaryOperator operator) =>
+      operator == BinaryOperator.times ||
+      operator == BinaryOperator.modulo ||
+      isAggressive &&
+          (operator == BinaryOperator.plus || operator == BinaryOperator.minus);
+
+  /// Returns true if we assume that [operator] always operators on numbers.
+  ///
+  /// This is always true for *, %, <, <=, >, and >=, and it's true for +, -,
+  /// ==, and != as long as the aggressive option is enabled.
+  bool operatesOnNumbers(BinaryOperator operator) =>
+      returnsNumbers(operator) ||
+      operator == BinaryOperator.lessThan ||
+      operator == BinaryOperator.lessThanOrEquals ||
+      operator == BinaryOperator.greaterThan ||
+      operator == BinaryOperator.greaterThanOrEquals ||
+      isAggressive &&
+          (operator == BinaryOperator.equals ||
+              operator == BinaryOperator.notEquals);
+
+  /// Returns true if [node] should be treated as division and migrated.
+  ///
+  /// Warns if division is allowed but it's unclear whether or not all types
+  /// are numeric.
+  bool shouldMigrate(BinaryOperationExpression node) {
+    if (!_isDivisionAllowed && onlySlash(node)) {
+      return false;
+    }
+    if (hasNonNumber(node)) return false;
+    if (_expectsNumericResult || _allNumeric(node)) return true;
+    warn("Could not determine whether this is division", node.span);
+    return false;
+  }
+
+  /// Returns true if [node] is entirely composed of number literals and slash
+  /// operations.
+  bool onlySlash(Expression node) {
+    if (node is NumberExpression) return true;
+    if (node is BinaryOperationExpression) {
+      return node.operator == BinaryOperator.dividedBy &&
+          onlySlash(node.left) &&
+          onlySlash(node.right);
+    }
+    return false;
+  }
+
+  /// Returns true if [node] is believed to always evaluate to a number.
+  bool _allNumeric(Expression node) {
```

Comment:
The naming difference between this and `hasNonNumber()` is confusing to me, since they seem like opposites. WDYT about calling them `_isDefinitelyNumber()` and `_isDefinitelyNotNumber()`?